src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Figure figure;
```

Comment:
Remove the class-level 'figure' variable and declare it locally within the 'getRandomFigure' method.